tgui/packages/tgui/interfaces/SupermatterMonitor.js

```diff
@@ -0,0 +1,147 @@
+import { Fragment } from 'inferno';
+import { useBackend } from '../backend';
+import { Section, Box, Button, Table, LabeledList, ProgressBar } from '../components';
+import { Window } from '../layouts';
+import { TableRow, TableCell } from '../components/Table';
+
+const powerToColor = power => {
+  if (power > 300) {
+    return 'bad';
+  } else if (power > 150) {
+    return 'average';
+  } else {
+    return 'good';
+  }
+};
+
+const temperatureToColor = temp => {
+  if (temp > 5000) {
+    return 'bad';
+  } else if (temp > 4000) {
+    return 'average';
+  } else {
+    return 'good';
+  }
+};
+
+const pressureToColor = pressure => {
+  if (pressure > 10000) {
+    return 'bad';
+  } else if (pressure > 5000) {
+    return 'average';
+  } else {
+    return 'good';
+  }
+};
+
+export const SupermatterMonitor = (props, context) => {
+  const { act, data } = useBackend(context);
+  if (data.active === 0) {
+    return (
+      <Window>
+        <Window.Content scrollable>
+          <Section title="Detected Supermatter Shards" buttons={
+            <Button
+              icon="sync"
+              content="Refresh"
+              onClick={() => act("refresh")}
+            />
+          }>
+            <Box m={1}>
+              {data.supermatters.length === 0 ? (
+                <h3>No shards detected</h3>
+              ) : (
+                <Table>
+                  <Table.Row header>
+                    <TableCell>Area</TableCell>
+                    <TableCell>Integrity</TableCell>
+                    <TableCell>Details</TableCell>
+                  </Table.Row>
+                  {data.supermatters.map(sm => (
+                    <TableRow key={sm}>
+                      <TableCell>{sm.area_name}</TableCell>
+                      <TableCell>{sm.integrity}%</TableCell>
+                      <TableCell>
+                        <Button
+                          icon="sign-in-alt"
+                          content="View"
+                          onClick={() => act('view', {
+                            view: sm.uid,
+                          })}
+                        />
+                      </TableCell>
+                    </TableRow>
+                  ))}
+                </Table>
+              )}
+            </Box>
+          </Section>
+        </Window.Content>
+      </Window>
+    );
+  } else {
+    return (
+      <Window>
+        <Window.Content>
+          <Section title="Crystal Status" buttons={
+            <Button
+              icon="caret-square-left"
+              content="Back"
+              onClick={() => act("back")}
+            />
+          }>
+            <LabeledList>
+              <LabeledList.Item label="Core Integrity">
+                <ProgressBar
+                  ranges={{
+                    good: [95, Infinity],
+                    average: [80, 94],
+                    bad: [-Infinity, 79],
+                  }}
```

Comment:
I'd make this a `const` at the top of the file.

Reply:
Progressbars must be done like this afaik